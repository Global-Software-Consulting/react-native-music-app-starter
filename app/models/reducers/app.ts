export interface AppState {
  musicList: music[];
  favoriteList:Array<object>;
  addFavoriteList:Array<object>;
  removeFavoriteList:Array<object>;
}
type music={
  id?: string;
  name?: string;
  url?: string;
  artwork?: string;
  artist?: string;
  duration?: number;
}