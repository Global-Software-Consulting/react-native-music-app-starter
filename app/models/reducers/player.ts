export interface PlayerState {
  isPlayer:Boolean;
  isPlayerPlay:Boolean;
  playerList: any;
  playList: playlist[];
}
type playlist={
  name:string;
  songs:songs[];
}
type songs={
  id?: string;
    title?: string;
    url?: string;
    artwork?: string;
    artist?: string;
}