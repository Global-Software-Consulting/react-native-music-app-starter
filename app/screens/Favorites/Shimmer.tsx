import React, { useState } from 'react';
import { View, FlatList, RefreshControl, ScrollView } from 'react-native';
import { tracks } from '../../data/tracks';
import useStyles from './styles';
import MusicCardShimmer from '../../components/Music/MusicCardShimmer';
import Header from '../../components/Header';
interface Itrack {
    id: string;
    url: string;
    title: string;
    artist: string;
    artwork: string;
    album: string;
    duration: number;
}
const FavoriteShimmer: React.FC<any> = () => {
    const Track: Itrack[] = tracks;
    const styles = useStyles();
    const [refreshing, setRefreshing] = useState<boolean>(false);
    const wait = (timeout: number) => {
        return new Promise((resolve) => setTimeout(resolve, timeout));
    };
    const onRefresh = React.useCallback(() => {
        setRefreshing(true);
        wait(2000).then(() => setRefreshing(false));
    }, []);
    const shimmerRenderItem = ({ item }: any) => (
        <View style={styles.Musiccontainer}>
            <MusicCardShimmer name={item.title} model={item.album} img={item.artwork} />
        </View>
    );
    return (
        <>
            <View style={styles.container}>
                <ScrollView
                    refreshControl={
                        <RefreshControl refreshing={refreshing} onRefresh={onRefresh} />
                    }>
                    <Header title="Liked Songs" />

                    <FlatList
                        contentContainerStyle={{ alignSelf: 'flex-start' }}
                        showsVerticalScrollIndicator={false}
                        showsHorizontalScrollIndicator={false}
                        numColumns={2}
                        data={Track}
                        keyExtractor={(item) => item.id}
                        renderItem={shimmerRenderItem}
                    />
                </ScrollView>
            </View>
        </>
    );
};

export default FavoriteShimmer;
