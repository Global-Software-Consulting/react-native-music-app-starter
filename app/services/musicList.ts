import { apiClient } from './client';
import ApiConfig from '../config/api-config';

export default function musicList() {
    return apiClient.get(ApiConfig.MUSIC_LIST);
}
