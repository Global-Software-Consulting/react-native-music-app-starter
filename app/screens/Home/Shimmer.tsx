import React, { useState } from 'react';
import { View, ScrollView, FlatList, RefreshControl, ToastAndroid } from 'react-native';
import useStyles from './styles';
import { useTranslation } from 'react-i18next';
import Header from '../../components/Header';
import i18n from "../../config/Languages/i18n";
import { tracks } from '../../data/tracks';
import { useSelector} from 'react-redux';
import MusicCardShimmer from '../../components/Music/MusicCardShimmer';
import {ILoading} from '../../models/reducers/loading';
const initI18n = i18n;

interface IState {
  loadingReducer: ILoading;
}

interface Itrack {

  id: string
  url: string
  title: string
  artist: string
  artwork: string
  album: string
  duration: number
};

const Home: React.FC<any> = (props): JSX.Element => {
  const [refreshing, setRefreshing] = useState<boolean>(false);
  const isLoading = useSelector((state:IState) => state.loadingReducer.isLoginLoading);

  const [listData, setListData] = useState<any>(tracks);
  const { t, i18n } = useTranslation();
  const Track: Itrack[] = tracks;
  const styles = useStyles();
  const wait = (timeout: number) => {
    return new Promise(resolve => setTimeout(resolve, timeout));
  }


  const PlayListRenderItem = ({ item }: any) => (

    <MusicCardShimmer
      name={item.title}
      model={item.album}
      img={item.artwork}
    />


  );
  const RecommendedRenderItem = ({ item }: any) => (

    <MusicCardShimmer
      name={item.title}
      model={item.album}
      img={item.artwork}
    />


  );

  return (
    <>
      <View style={styles.container}>
        <ScrollView
          refreshControl={<RefreshControl
            refreshing={isLoading}
          />}
        >
          <Header title="Recommended for you" />
          <FlatList
            contentContainerStyle={{ alignSelf: 'flex-start' }}
            horizontal={true}
            showsVerticalScrollIndicator={false}
            showsHorizontalScrollIndicator={false}
            data={Track}
            scrollEventThrottle={2}
            keyExtractor={(item) => item.id}
            renderItem={RecommendedRenderItem}


          />
          <Header title="My Playlist" />

          <FlatList
            contentContainerStyle={{ alignSelf: 'flex-start' }}
            horizontal={true}
            showsVerticalScrollIndicator={false}
            showsHorizontalScrollIndicator={false}
            data={Track}
            keyExtractor={(item) => item.id}
            // renderItem={renderItemsss}
            renderItem={PlayListRenderItem}


          />
        </ScrollView>

      </View>
    </>
  );
};

export default Home;
