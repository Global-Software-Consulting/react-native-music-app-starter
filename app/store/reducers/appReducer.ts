/* Login Reducer
 * handles login states in the app
 */
import createReducer from 'lib/createReducer';
import * as types from 'store/actions/types';

import { IAppState } from 'models/reducers/app';
import {
  ILoginRequestState,
  ILoginResponseState,
} from 'models/actions/login';
const initialState: IAppState = {
  musicList:[]
};

export const appReducer = createReducer(initialState, {
 
  [types.MUSIC_LIST_RESPONSE](state: IAppState, action) {
    return {
      ...state,
      musicList: action.response,
      
    };
  },
 
});
