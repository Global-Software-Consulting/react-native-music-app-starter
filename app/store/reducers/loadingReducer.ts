/**
 * Loading reducer made separate for easy blacklisting
 * Avoid data persist
 */
import createReducer from 'lib/createReducer';
import * as types from 'store/actions/types';
import { ILoading } from 'models/reducers/loading';

const initialState: ILoading = {
  isLoginLoading: false,
};

export const loadingReducer = createReducer(initialState, {
  [types.MUSIC_ENABLE_LOADER](state: ILoading) {
    return { ...state, isLoginLoading: true };
  },
  [types.MUSIC_DISABLE_LOADER](state: ILoading) {
    return { ...state, isLoginLoading: false };
  },
});
