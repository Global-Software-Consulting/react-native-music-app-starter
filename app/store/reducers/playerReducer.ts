/* Login Reducer
 * handles login states in the app
 */
import createReducer from 'lib/createReducer';
import * as types from 'store/actions/types';
import { IPlayerState } from '../../models/reducers/player';

const initialState: IPlayerState = {
  playerList: [],
  isPlayer: false,
  isPlayerPlay: false,
  playList: [],

};

export const playerReducer = createReducer(initialState, {

  //list of selected item that is playing
  [types.PLAYER_LIST_REQUEST](state: IPlayerState, action: any) {
    return {
      ...state,
      playerList: action.payload,
    };
  },

  //Flag for showing bottom player
  [types.ISPLAYER_SHOWN](state: IPlayerState, action: any) {
    // console.log("action.response",action.response);

    return {
      ...state,
      isPlayer: action.payload,

    };
  },

  //Flag for showing playing state in bottom and full screen
  [types.MUSIC_PLAYER_PLAY](state: IPlayerState, action: any) {
    console.log("action.response", action.response);

    return {
      ...state,
      isPlayerPlay: action.payload,

    };
  },

  //Playlist
  [types.ADD_PLAY_LIST](state: IPlayerState, action: any) {
    return {
      ...state,
      playList: action.payload,
    }
  },
  [types.UPDATE_PLAY_LIST](state: IPlayerState, action: any) {
    return {
      ...state,
      playList: action.payload,
    };
  },
  [types.DELETE_PLAY_LIST](state: IPlayerState, action: any) {
    return {
      ...state,
      playList: action.payload,
    }
  },
  [types.DELETE_SONG_IN_PLAYLIST](state: IPlayerState, action: any) {
    return {
      ...state,
      playList: action.payload,
    }
  },


});
