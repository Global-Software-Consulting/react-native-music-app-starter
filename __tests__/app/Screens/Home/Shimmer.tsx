/**
 * @format
 */
import 'react-native';
import React from 'react';
import Shimmer from '../../../../app/screens/Home/Shimmer';
import * as reactRedux from 'react-redux';
// Note: test renderer must be required after react-native.
import { render } from '@testing-library/react-native';
import { ReducerState } from '../../../../app/models/reducers/index'
import musicList from 'app/services/musicList';
jest.setTimeout(15000);

jest.mock("react-native-responsive-screen", () => {
    return {
        heightPercentageToDP: jest.fn()
    }


});
jest.mock('@react-navigation/native', () => {
    return {
        useIsFocused: jest.fn(),
        useNavigation: jest.fn()
    }


});
jest.mock('react-i18next', () => {
    return {
        useTranslation: () => {
            return {
                t: jest.fn(),
            };
        },
    };
});

jest.mock('react-native-iphone-x-helper', () => ({
    getStatusBarHeight: jest.fn(),
    getBottomSpace: jest.fn(),
}));
jest.mock('react-redux', () => ({
    useDispatch: jest.fn(),
    useSelector: jest.fn()

}));
describe("group tesing goes here", () => {
    test('renders correctly', () => {
        // const musicListState: ReducerState = {
        //     appReducer: musicList,
        // };
        // const musicListMock = jest
        //     .spyOn(reactRedux, 'useSelector')
        //     .mockImplementation((cb) => cb(musicListState));
        // musicListMock.mockClear();

        const tree = render(<Shimmer />);
        expect(tree).toMatchSnapshot();
    });
})

