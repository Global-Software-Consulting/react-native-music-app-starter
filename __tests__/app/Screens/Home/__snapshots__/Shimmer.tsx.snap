// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`group tesing goes here renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#03dac4",
      "flex": 1,
      "paddingBottom": "15%",
    }
  }
>
  <RCTScrollView
    refreshControl={<RefreshControlMock />}
  >
    <RCTRefreshControl />
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#03dac4",
            "flexDirection": "row",
            "padding": 15,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Object {
                  "fontSize": undefined,
                  "fontWeight": "bold",
                },
              ]
            }
          />
        </View>
      </View>
      <RCTScrollView
        contentContainerStyle={
          Object {
            "alignSelf": "flex-start",
          }
        }
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={2}
        showsHorizontalScrollIndicator={false}
        showsVerticalScrollIndicator={false}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#03dac4",
            "flexDirection": "row",
            "padding": 15,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Object {
                  "fontSize": undefined,
                  "fontWeight": "bold",
                },
              ]
            }
          />
        </View>
      </View>
      <RCTScrollView
        contentContainerStyle={
          Object {
            "alignSelf": "flex-start",
          }
        }
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsHorizontalScrollIndicator={false}
        showsVerticalScrollIndicator={false}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
    </View>
  </RCTScrollView>
</View>
`;
